include/llvm/Support/WinResults.h

```diff
@@ -2,9 +2,9 @@
 #define LLVM_SUPPORT_WINRESULTS_H
 
 #define S_OK 0x00000000
+#define E_FAIL -1
```

Comment:
I'm curious why this is needed?

Reply:
This is because of
```cpp
#define SUCCEEDED(hr) (((HRESULT)(hr)) >= 0)
```
We must have a negative number to indicate failure, otherwise existing code won't work.
I wonder if we should change all the E_* macros to negative numbers.

Reply:
Actually `E_FAIL` and others are defined at https://msdn.microsoft.com/en-us/library/windows/desktop/aa378137(v=vs.85).aspx. They are standard values. We should probably keep them consistent with Windows defines.

I think the problem is `HRESULT` is defined to be `long` for some reason. We should use `int32_t` then the above `E_*` values will be correctly recognized as negative values.

Reply:
I know that these are defined values for Windows... but according to https://msdn.microsoft.com/en-us/library/windows/desktop/aa383751(v=vs.85).aspx
HRESULT is of type `long`. That's why i was not sure whether to change hresult to `int` or to change the enum value to -1...
(the enum value itself doesn't matter as long as it is unique, because Linux doesn't recognize it anyway, right?)

Reply:
@antiagainst does `long a = 0x80004005` evaluate to a positive number and `int a = 0x80004005` evaluate to a negative number?

Reply:
I think so. I guess it's because compiler difference. Note that  `0x80004005` has its most significant bit set. MSVC interprets `0x80004005` with signed expansion but GCC/Clang does zero expansion? So either we change to `#define` `HRESULT` as `int32_t`, or we expand all the `E_*` values to true 64-bit. Both are fine for me.

Reply:
Note that Long is guaranteed to be 32 bits on Windows, but it is 64 bits on Linux. The other place I ran into problems here is the LARGE_INTEGER union. My vote is for int32_t. It deviates the least from the code's original expectations. Currently HResult is defined to long. So it's 64 bits. That's why the 0x8.... values aren't coming out as negative.

Reply:
Thanks for the clarification. I'm trying to switch the definition of HRESULT to int32_t for Linux. However, I'm not seeing the the behavior that we expect (I think it is still being evaluated as a positive number :-( ). I'm debugging that now.

Reply:
Ah. Found the issue. There's a duplicate definition of HRESULT in Global.h